### PR TITLE
Introduce ProjectAPI 

### PR DESCRIPTION
**Pull request by J535D165**
This PR introduces a new ProjectAPI. This simplifies the REST API and Simulation feature.

### CODE REVIEW

File: asreview/project.py [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+        except AttributeError:
+            # file seems to be closed, do nothing
+            pass
+
+
+class ASReviewProject():
+
+    def __init__(self, project_path, project_id=None):
+        self.project_path = project_path
+        self.project_id = project_id
```
Comment (J535D165) [inline asreview/project.py:206]: Remove and create on `ASReviewProject.create`.

File: asreview/project.py [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+    Returns
+    -------
+    SQLiteState
+    """
+
+    # Unzip the ASReview data if needed.
+    if zipfile.is_zipfile(asreview_file_or_dir) and Path(asreview_file_or_dir).suffix == ".asreview":
+
+        if not read_only:
+            raise ValueError("ASReview files do not support not read only files.")
```
Comment (PeterLombaers) [inline asreview/project.py:174]: What is happening here?

File: asreview/project.py [hunk 3]
```diff
@@ -0,0 +1,669 @@
...
+            raise ValueError("Project name should be at least 3 characters.")
+
+        if project_path.is_dir():
+            raise IsADirectoryError(
+                f'Project folder {project_path} already exists.')
+
+        try:
+            project_path.mkdir(exist_ok=True)
+            get_data_path(project_path).mkdir(exist_ok=True)
+            Path(project_path, PATH_FEATURE_MATRICES).mkdir(exist_ok=True)
```
Comment (PeterLombaers) [inline asreview/project.py:266]: get_feature_matrices_path

File: asreview/webapp/utils/project.py [hunk 4]
```diff
@@ -1,564 +0,0 @@
...
-            labels = as_data.labels[labeled_indices].tolist()
-            labeled_record_ids = as_data.record_ids[labeled_indices].tolist()
-
-            # add the labels as prior data
-            state.add_labeling_data(
-                record_ids=labeled_record_ids,
-                labels=labels,
-                notes=[None for _ in labeled_record_ids],
-                prior=True
-            )
```
Comment (terrymyc) [inline asreview/webapp/utils/project.py:303]: @J535D165 This was not completely ported to the new `project.py`. This condition `project_info["mode"] != PROJECT_MODE_SIMULATE` is missing.

File: asreview/project.py [hunk 5]
```diff
@@ -0,0 +1,659 @@
...
+                labels = as_data.labels[labeled_indices].tolist()
+                labeled_record_ids = as_data.record_ids[
+                    labeled_indices].tolist()
+
+                # add the labels as prior data
+                state.add_labeling_data(
+                    record_ids=labeled_record_ids,
+                    labels=labels,
+                    notes=[None for _ in labeled_record_ids],
+                    prior=True)
```
Comment (terrymyc) [inline asreview/project.py:434]: The labels in the dataset should not be added to the state file if the mode is simulation.

**Comment by J535D165** [comment]
There are some things still broken (unit tests pass), I will resolve them in future PRs.

**Pull request merged**